test/OpenTelemetry.Instrumentation.Runtime.Tests/RuntimeMetricsTests.cs

```diff
@@ -34,9 +37,156 @@ public void RuntimeMetricsAreCaptured()
                  .AddInMemoryExporter(exportedItems)
                 .Build();
 
+            // The process.runtime.dotnet.exception.count metrics are only available after an exception has been thrown post OpenTelemetry.Instrumentation.Runtime initialization.
+            try
+            {
+                throw new Exception("Oops!");
+            }
+            catch (Exception)
+            {
+                // swallow the exception
+            }
+
             meterProvider.ForceFlush(MaxTimeToAllowForFlush);
             Assert.True(exportedItems.Count > 1);
             Assert.StartsWith(MetricPrefix, exportedItems[0].Name);
+
+            var assembliesCountMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.assemblies.count");
+            Assert.NotNull(assembliesCountMetric);
+
+            var exceptionsCountMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.exceptions.count");
+            Assert.True(GetValue(exceptionsCountMetric) >= 1);
+        }
+
+        [Fact]
+        public void GcMetricsTest()
+        {
+            var exportedItems = new List<Metric>();
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                 .AddRuntimeInstrumentation()
+                 .AddInMemoryExporter(exportedItems)
+                .Build();
+
+            GC.Collect(1);
+
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+
+            var gcCountMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.gc.collections.count");
+            Assert.NotNull(gcCountMetric);
+
+#if NETCOREAPP3_1_OR_GREATER
+            var gcAllocationSizeMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.gc.allocations.size");
+            Assert.NotNull(gcAllocationSizeMetric);
+#endif
+
+#if NET6_0_OR_GREATER
+            var gcCommittedMemorySizeMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.gc.committed_memory.size");
+            Assert.NotNull(gcCommittedMemorySizeMetric);
+
+            var gcHeapSizeMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.gc.heap.size");
+            Assert.NotNull(gcHeapSizeMetric);
+
+            if (Environment.Version.Major >= 7)
+            {
+                var gcHeapFragmentationSizeMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.gc.heap.fragmentation.size");
+                Assert.NotNull(gcHeapFragmentationSizeMetric);
```

Comment:
I manually tested this with .NET 7.0 preview: installing Visual Studio 17.3.0 Preview 4.0 and adding `net7.0` in `TargetFrameworks` tag of OpenTelemetry.Instrumentation.Runtime.Tests.csproj file.

Because .NET 7.0 is still in Preview status, I'm not making the change for csproj and CI process in this PR.